src/legacy/ui/public/courier/search_source/search_source.js

```diff
@@ -551,12 +551,13 @@ export function SearchSourceProvider(Promise, Private, config) {
           flatData.body = flatData.body || {};
 
           const computedFields = flatData.index.getComputedFields();
+
           flatData.body.stored_fields = computedFields.storedFields;
           flatData.body.script_fields = flatData.body.script_fields || {};
-          flatData.body.docvalue_fields = flatData.body.docvalue_fields || [];
-
           _.extend(flatData.body.script_fields, computedFields.scriptFields);
-          flatData.body.docvalue_fields = _.union(flatData.body.docvalue_fields, computedFields.docvalueFields);
+
+          const defaultDocValueFields = computedFields.docvalueFields ? computedFields.docvalueFields : [];
+          flatData.body.docvalue_fields = flatData.body.docvalue_fields || defaultDocValueFields;
```

Comment:
I'm a bit hesitant to exclude the `computedFields.docvalueFields` when other docvalue_fields are explicitly provided. While it might be unnecessary in the maps example you provided, there could be other code relying on this behavior. Unless the extra data in the payload is causing a specific issue, I wouldn't say the bandwidth savings are worth the risk of a breakage.

Reply:
When you are pulling back 10000 hits on a slower connection it has a performance effect. We lazy load the properties for the tooltips for this exact reason. The smaller the payload makes a big difference.

Reply:
10000 ISO date strings is only 80KB when gzipped. Not completely trivial maybe, but your connection would have to be pretty slow to notice a difference. Not a big enough difference IMO to change the workings of an extremely old function that is used all throughout the code base.

Reply:
If you grep for `docvalue_fields` in either src or x-pack, it looks like no one is currently using SearchSource with an initial search context of docvalue_fields so this change will not effect any existing consumers of SearchSource.

Reply:
alright, sounds good 👍